### ISSUE TITLE
mhx2 Exporter - 'int' object has no attribute 'object'

### ISSUE THREAD

**Issue by wutayng** [makehumancommunity/makehuman#75]
I am equipping different assets with both .mhpxy and .mhclo files (mostly successfully). However, this error I have not been able to fix - it happens every time I try to export my human with this particular asset. The asset is equipped successfully and the error message occurs upon .mhx2 export. Any help would be much appreciated! Here is the full log, starting when I execute my script:

---
hiding task Logs  

showing task Execute  

Loading binary proxy /Users/username/data/makeHumanAssets/eyebrows/eyebrow011/eyebrow011.mhpxy.

Loading proxy file: /Users/username/data/makeHumanAssets/eyebrows/eyebrow011/eyebrow011.mhpxy.

Loading material from file /Users/username/data/makeHumanAssets/eyebrows/eyebrow011/eyebrow011.mhmat

Loading binary mesh /Users/username/data/makeHumanAssets/eyebrows/eyebrow011/eyebrow011.npz.

0 faces masked for eyebrow011.obj

0 faces masked for base.obj

Write MHX2 file /Users/username/MakeHuman/v1/exports/testing

'int' object has no attribute 'object'
Traceback (most recent call last):
  File "plugins/7_scripting.py", line 823, in executeScript
  File "<string>", line 82, in <module>
  File "/Users/username/makehumans/tests/test_functions.py", line 104, in testAssets
  File "/Users/username/makehumans/tests/test_functions.py", line 49, in equip_save_and_delete
  File "/Users/username/makehumans/lib/core.py", line 339, in exportMHX2
  File "plugins/1_mhapi/_exports.py", line 83, in exportAsMHX2
  File "plugins/9_export_mhx2/__init__.py", line 55, in export
  File "plugins/9_export_mhx2/mh2mhx2.py", line 54, in exportMhx2
  File "./apps/human.py", line 358, in getObjects
  File "./apps/human.py", line 348, in getProxyObjects
AttributeError: 'int' object has no attribute 'object'
---

**Comment by Aranuvir** [comment]
I don't know what you are doing here. Running MHX2 over the scripting plug-in is not the recommended way and 'v1' indicates that you are using an outdated version of MakeHuman. I don't think any of the new plugins will interact correctly with it.

Suggesting to close this task.

**Comment by wutayng** [comment]
Thanks your your help! I am running on macOS, so I am using the stable release v1.1.1. I am under the impression that building from source is not available on mac, is that incorrect?

I was unaware that MHX2 was not meant to be run over the scripting plug in. Are you suggesting there is there a more proper way to export MHX2 (not using GUI)? Any pointers to documentation I may have missed would be great - thanks again!

**Comment by Aranuvir** [comment]
Sorry I might be missing something. Are you running your own scripts on the scripting tab?
I do not get this error when I export the base human with eyebrows011. Can you retry with deleting the mhpxy/npz (these are just some sort of cache files)?
And if this does not help, can you retry with a more recent version of MH? You don't need to build the application, it's pure Python code. To run it, install Python and within the interpreter you need to install numpy, PyQt5 and PyOpenGL. Even if MH 1.2.0 is still considered alpha it is rather stable...

**Comment by wutayng** [comment]
Thanks for your help, I figured out the problem. In the scripting tab, I was adding different assets, exporting as mhx2, removing the asset, repeat.

Something about my process of unequpping assets via the following was causing the mhx2 exporter to fail in the above way.
```
human.removeClothesProxy(pxyfile.uuid)
human.setHairProxy(0)
G.app.mhapi.assets.unequipAllClothes()
```

Instead of this, for each iteration to clear the human I simply did
```
G.app._resetHuman()
```
And now the exporter runs without error.
